tests/check/check_test.go

```diff
@@ -33,6 +37,130 @@ func TestCheckMySQL(t *testing.T) {
+		TypeDefinitions: typedefs,
+	})
+	require.NoError(t, err)
+
+	authorizationModelID := writeModelResp.GetAuthorizationModelId()
+
+	_, err = client.Write(context.Background(), &openfgav1.WriteRequest{
+		StoreId: storeID,
+		Writes: &openfgav1.TupleKeys{
+			TupleKeys: []*openfgav1.TupleKey{
+				tuple.NewTupleKey("document:1", "viewer", "user:anne"),
+			},
+		},
+	})
+	require.NoError(t, err)
+
+	type test struct {
+		_name           string
+		req             *openfgav1.CheckRequest
+		expectedContext map[string]interface{}
+	}
+
+	tests := []test{
+		{
+			_name: "check_success",
+			req: &openfgav1.CheckRequest{
+				AuthorizationModelId: authorizationModelID,
+				StoreId:              storeID,
+				TupleKey:             tuple.NewTupleKey("document:1", "viewer", "user:anne"),
+			},
+			expectedContext: map[string]interface{}{
+				"grpc_service":           "openfga.v1.OpenFGAService",
+				"grpc_method":            "Check",
+				"grpc_type":              "unary",
+				"grpc_code":              int32(0),
+				"raw_request":            fmt.Sprintf(`{"store_id":"%s","tuple_key":{"object":"document:1","relation":"viewer","user":"user:anne"},"contextual_tuples":null,"authorization_model_id":"%s","trace":false}`, storeID, authorizationModelID),
+				"raw_response":           `{"allowed":true,"resolution":""}`,
+				"authorization_model_id": authorizationModelID,
+				"store_id":               storeID,
+			},
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test._name, func(t *testing.T) {
+			_, _ = client.Check(context.Background(), test.req)
+
+			filteredLogs := logs.Filter(func(e observer.LoggedEntry) bool {
+				if e.Message == "grpc_req_complete" {
+					for _, ctxField := range e.Context {
+						if ctxField.Equals(zap.String("grpc_method", "Check")) {
+							return true
+						}
+					}
+				}
+
+				return false
+			})
+
+			expectedLogs := filteredLogs.All()
+			require.Len(t, expectedLogs, 1)
+
+			fields := expectedLogs[len(expectedLogs)-1].ContextMap()
+
+			require.Equal(t, test.expectedContext["grpc_service"], fields["grpc_service"])
+			require.Equal(t, test.expectedContext["grpc_method"], fields["grpc_method"])
+			require.Equal(t, test.expectedContext["grpc_type"], fields["grpc_type"])
+			require.Equal(t, test.expectedContext["grpc_code"], fields["grpc_code"])
+			require.JSONEq(t, test.expectedContext["raw_request"].(string), string(fields["raw_request"].(json.RawMessage)))
+			require.JSONEq(t, test.expectedContext["raw_response"].(string), string(fields["raw_response"].(json.RawMessage)))
+			require.Equal(t, test.expectedContext["authorization_model_id"], fields["authorization_model_id"])
+			require.Equal(t, test.expectedContext["store_id"], fields["store_id"])
+			require.GreaterOrEqual(t, fields["datastore_query_count"], int64(1))
+			require.LessOrEqual(t, fields["datastore_query_count"], int64(2))
+			require.NotEmpty(t, fields["peer.address"])
+			require.NotEmpty(t, fields["request_id"])
```

Comment:
If you enable tracing, there is one more assertion you can do: `require.NotEmpty(t, fields["trace_id"])`